### PR TITLE
Enhancement/1610/method to force compact symbol list

### LINKED ISSUE #1610: Add method to forcibly compact the symbol list, and throw a useful error message if it fails
`list_symbols` compacts by default if there are too many symbol list cache keys and the user has write permissions on the library. If this compaction fails for any reason though, the call will still succeed and return the correct result, so there is no way to identify a failed compaction right now.

### PR DESCRIPTION
**Pull request by alexowens90**
#### Reference Issues/PRs
Closes #1610 

Considered augmenting `list_symbols` API to make this an option, but there are already a lot of arguments (particularly in the `NativeVersionStore` version) that this would not be compatible with, so added another method instead.

### CODE REVIEW

File: cpp/arcticdb/util/storage_lock.hpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ class StorageLock {
         OnExit x{[that=this] () {
             that->mutex_.unlock();
         }};
-        if(!ref_key_exists(store) || ttl_expired(store)) {
+        if(!ref_key_exists(store) || !ttl_not_expired(store)) {
```
Comment (vasil-pashov) [inline cpp/arcticdb/util/storage_lock.hpp:119]: I find double negatives hard to read.

Comment (willdealtry) [reply cpp/arcticdb/util/storage_lock.hpp:119]: If I were refactoring I would probably replace both sides of the || with a single-line method whose name explains what is being checked `if(no_lock_object_present() || lock_object_within_ttl()`

**Review by willdealtry — APPROVED** [review]

File: python/arcticdb/version_store/library.py [hunk 2]
```diff
@@ -1706,6 +1706,20 @@ def reload_symbol_list(self):
...
+
+
+        Raises
+        ------
+        PermissionException
+            Library has been opened in read-only mode
+        InternalException
+            Storage lock required to compact the symbol list could not be acquired
+        """
+        self._nvs.compact_symbol_list()
```
Comment (jamesmunro) [inline python/arcticdb/version_store/library.py:1721]: Can it return something to indicate whether it did anything to storage, or how much compaction it did?

Comment (alexowens90) [reply python/arcticdb/version_store/library.py]: The method will always compact the symbol list cache down to a single key or throw. I've updated to return the number of symbol list keys prior to the compaction though

**Pull request merged**